app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -30,6 +29,10 @@ class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
 
     attr_accessor :client
 
+    def middleware_server_groups
```

Comment:
With this refactor, inventory is being saved by `ManagerRefresh::SaveCollection::TopologicalSort#save_collections` (https://git.io/vSscY). Looks like this assumes that the manager has a direct relation with all its inventory collections, although references between entities are allowed. However, in Hawkular relationships are EMS -> Domains -> Server Groups. Yes, its a little hacky, but I found no other way but to add this to mimic the behavior of a relation.

May be @Ladas can comment if there is a way to remove this.

This is the question I wanted to ask (commented to you on Thursday on our meeting).